File: test/Zorglub.Tests.CSharp/Time/Simple/CuidTests.cs
// SPDX-License-Identifier: BSD-3-Clause
// Copyright (c) 2020 Narvalo.Org. All rights reserved.

namespace Zorglub.Time.Simple;

public static class CuidTests
{
    [Theory]
    [InlineData(Cuid.Zoroastrian + 1)]
    [InlineData(Cuid.MinUser)]
    [InlineData(Cuid.Max)]
    [InlineData(Cuid.Invalid)]
    internal static void IsFixed_False(Cuid id) =>
        Assert.False(id.IsFixed());

    [Theory, MemberData(nameof(EnumDataSet.CalendarIdData), MemberType = typeof(EnumDataSet))]
    public static void IsFixed_True(CalendarId id) =>
        Assert.True(((Cuid)id).IsFixed());
}

File: test/Zorglub.Tests/Tests/Simple/CuidTests.fs
// SPDX-License-Identifier: BSD-3-Clause
// Copyright (c) Tran Ngoc Bich. All rights reserved.

module Zorglub.Tests.Simple.CuidTests

open Zorglub.Testing
open Zorglub.Testing.Data

open Zorglub.Time.Simple

open Xunit

open Zorglub.Time.FSharpExtensions

let unfixedCuidData = EnumDataSet.UnfixedCuidData
let fixedCuidData = EnumDataSet.FixedCuidData

[<Theory; MemberData(nameof(unfixedCuidData))>]
let ``IsFixed() returns false when the id is not fixed`` (cuid: Cuid) =
    isFixed cuid |> nok

[<Theory; MemberData(nameof(fixedCuidData))>]
let ``IsFixed() returns true when the id is fixed`` (cuid: Cuid) =
    isFixed cuid |> ok
